rust/src/traits/samplers/psrn/mod.rs

```diff
@@ -142,38 +49,51 @@ impl GumbelPSRN {
     }
 }
 
+pub trait ODPRound: ErrorBounds {
+    const UBIG: UBig;
+    type Complement: ODPRound<Complement = Self>;
+}
+
+impl ODPRound for Down {
+    const UBIG: UBig = UBig::ZERO;
+    type Complement = Up;
+}
+
+impl ODPRound for Up {
+    const UBIG: UBig = UBig::ONE;
+    type Complement = Down;
+}
+
+// fn psrn_value<TI: PSRN<Edge = Rational>, TO: CastInternalRational + PartialEq>(
+//     psrn: &mut TI,
+// ) -> Fallible<TO> {
+//     while TO::from_rational(psrn.edge(Lower)?) != TO::from_rational(psrn.edge(Upper)?) {
+//         psrn.refine()?;
+//     }
+//     Ok(TO::from_rational(psrn.edge(Lower)?))
+// }
+
 #[cfg(test)]
-mod test {
+pub mod test {
     use super::*;
 
-    #[test]
-    fn test_sample_gumbel_interval_progression() -> Fallible<()> {
-        let mut gumbel = GumbelPSRN::new(RBig::ZERO, RBig::ONE);
-        for _ in 0..10 {
-            println!(
-                "{:?}, {:?}, {}",
-                gumbel.value::<Down>()?.to_f64(),
-                gumbel.value::<Up>()?.to_f64(),
-                gumbel.precision
-            );
-            gumbel.refine()?;
-        }
-        Ok(())
-    }
+    pub fn test_progression<RV: PSRN>(
```

Comment:
```suggestion
    #[test]
    pub fn test_progression<RV: PSRN>(
```
left off by mistake?

A one-liner to find other outliers:
```
grep 'fn test' -B1 -r -n src | egrep '^\S+-\d+-' | grep -v '#\[test\]'
src/core/mod.rs-418-    #[cfg(feature = "ffi")]
src/transformations/quantile_score_candidates/mod.rs-307-
src/traits/cast/mod.rs-394-    #[ignore]
src/traits/mod.rs-68-/// use opendp::traits::Primitive;
src/traits/mod.rs-111-/// use std::collections::HashSet;
src/traits/mod.rs-138-/// use opendp::traits::Number;
src/traits/mod.rs-223-/// use std::collections::HashSet;
src/traits/mod.rs-253-/// use opendp::traits::Float;
src/traits/samplers/uniform/mod.rs-207-    #[ignore]
src/traits/samplers/uniform/mod.rs-221-    #[ignore]
src/traits/samplers/uniform/mod.rs-235-    #[ignore]
src/traits/samplers/mod.rs-135-
src/traits/samplers/psrn/mod.rs-79-
src/ffi/util.rs-449-    #[rustfmt::skip]
src/ffi/util.rs-463-    #[rustfmt::skip]
src/data/mod.rs-114-    // #[should_panic]
src/data/mod.rs-120-
src/measurements/laplace_threshold/mod.rs-156-    #[cfg(feature = "partials")]
```